--- front-2-fragment-annotated.mlir
+++ front-2-scale-annotated.mlir
@@ -45,7 +45,7 @@
               %18 = tileflow.mask_fill %10 active(%17 : !tileflow.mask<bf16>) policy zero {tileflow.realization_engine = #tileflow.engine<dm>} : !d2mtile.tile<bf16> -> !d2mtile.tile<bf16>
               %19 = tileflow.col_mask symbols(%16) set #set : !tileflow.mask<bf16>
               %20 = tileflow.mask_fill %14 active(%19 : !tileflow.mask<bf16>) policy zero {tileflow.realization_engine = #tileflow.engine<dm>} : !d2mtile.tile<bf16> -> !d2mtile.tile<bf16>
-              %21 = d2mtile.binary tile_mul %18, %7 -> "source_ssa" : !d2mtile.tile<bf16>
+              %21 = d2mtile.binary tile_mul %18, %7 -> "source_ssa" {tileflow.realization_engine = #tileflow.engine<dm>} : !d2mtile.tile<bf16>
               %22 = d2mtile.binary tile_mul %21, %20 -> "source_ssa" : !d2mtile.tile<bf16>
               %23 = d2mtile.binary tile_sub %15, %22 -> "source_ssa" : !d2mtile.tile<bf16>
               tileplan.write %23, %1[%arg7, %arg8] : !d2mtile.tile<bf16>, !tileplan.shard_view
